taskcluster/mozillavpn_taskgraph/transforms/beetmover.py

```diff
@@ -23,23 +22,72 @@ def add_beetmover_worker_config(config, tasks):
         build_id = config.params["moz_build_date"]
         build_type = task["attributes"]["build-type"]
         build_os = os.path.dirname(build_type)
-        app_version = config.params["version"]
-        candidates_path = os.path.join(
-            "pub", app_name, "candidates", app_version, build_id, build_os
-        )
-        destination_paths = [candidates_path]
+        shipping_phase = config.params.get("shipping_phase", "")
+
+        if config.params["version"]:
+            app_version = config.params["version"]
+        elif "releases" in config.params["head_ref"]:
+            app_version = config.params["head_ref"].split("/")[-1]
+        else:
+            app_version = ""  # addons are not versioned
+
+        destination_paths = []
+
+        if build_type == "addons/opt":
+            destination_paths.append(
+                os.path.join(
+                    "pub",
+                    "vpn",
+                    "addons",
+                    "releases" if shipping_phase.startswith("ship") else "candidates",
+                    build_id,
+                )
+            )
+        else:
+            destination_paths.append(
+                os.path.join(
+                    "pub",
+                    "vpn",
+                    "candidates",
+                    f"{app_version}-candidates",
+                    f"build{build_id}",
+                    build_os,
+                )
+            )
+
+        if shipping_phase == "ship-addons":
+            destination_paths.append(
+                os.path.join(
+                    "pub",
+                    "vpn",
+                    "addons",
+                    "releases",
+                    "latest",
+                )
+            )
+
         archive_url = (
             "https://ftp.mozilla.org/" if is_relpro else "https://ftp.stage.mozaws.net/"
         )
-        task_description = f"This {worker_type} task will upload a {build_os} release candidate for v{app_version} to {archive_url}{candidates_path}/"
-        branch = config.params["head_ref"]
+
+        if build_type == "addons/opt" and task["name"] == "addons-bundle":
+            addons = set(os.listdir("addons"))
+            addons.remove("examples")
+            for addon in addons:
+                task["attributes"]["release-artifacts"].append(
+                    {
+                        "type": "file",
+                        "name": f"public/build/{addon}.rcc",
+                        "path": f"/builds/worker/artifacts/{addon}.rcc",
+                    }
+                )
```

Comment:
Also not a blocker, but imo it's nice to split transforms up into smaller logical pieces. This bit in particular looks like a good candidate for a separate transform.